crates/polars-core/src/datatypes/any_value.rs

```diff
@@ -536,8 +536,40 @@ impl<'a> AnyValue<'a> {
             (AnyValue::Float64(v), DataType::Boolean) => AnyValue::Boolean(*v != f64::default()),
 
             // to string
-            (av, DataType::String) => {
-                AnyValue::StringOwned(format_smartstring!("{}", av.extract::<i64>()?))
+            (AnyValue::UInt8(_v), DataType::String) => {
```

Comment:
I don't think we need full code branches for every type.

We can check if the `av` is float and in that case extract `f64`. We should add a case for `u64`. For the other cases we can keep `i64` extraction.

Reply:
Ok I can do, but as I was testing edge cases I found this issue in polars 1.2.1:

```
>>> pl.select(pl.lit(-3, dtype=pl.UInt32).cast(pl.Utf8))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/eric/miniconda3/envs/polar/lib/python3.12/site-packages/polars/functions/lazy.py", line 1899, in select
    return pl.DataFrame().select(*exprs, **named_exprs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/eric/miniconda3/envs/polar/lib/python3.12/site-packages/polars/dataframe/frame.py", line 8618, in select
    return self.lazy().select(*exprs, **named_exprs).collect(_eager=True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/eric/miniconda3/envs/polar/lib/python3.12/site-packages/polars/lazyframe/frame.py", line 1942, in collect
    return wrap_df(ldf.collect(callback))
                   ^^^^^^^^^^^^^^^^^^^^^
polars.exceptions.InvalidOperationError: conversion from `i32` to `u32` failed in column 'literal' for 1 out of 1 values: [-3]
```

From my understand this should be allowed? Should I fill another issue or fix it then this PR?